### PR TITLE
Migrate login and authentication in Back Office to Symfony

### LINKED ISSUE #36002: Migrate login form, logout and handle authorization via Symfony, remove Symfony layout feature flag
https://github.com/PrestaShop/PrestaShop/pull/35983

### PR DESCRIPTION
**Pull request by jolelievre**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Migrate login and authentication in Back Office to Symfony
| Type?             | new feature
| Category?         | BO
| BC breaks?        | yes
| Deprecations?     | yes
| How to test?      | CI tests and UI tests green<br>Manual checks:<br>- check login to BO<br>- check logout from BO<br>- check password reset feature<br>- check permission are consistent with assigned profile<br>- check profile edition
| UI Tests          | https://github.com/jolelievre/ga.tests.ui.pr/actions/runs/8921326681
| Fixed issue or discussion?     | Fixes #36002
| Related PRs       | https://github.com/PrestaShop/ga.tests.ui.pr/pull/67 GA UI test tool updated to remove the option about symfony layout that is no forced and the only possibility
| Sponsor company   | ~

### CODE REVIEW

**Review by Progi1984 — DISMISSED** [review]
Ok for UI Tests

**Review by Progi1984 — APPROVED** [review]
Ok for UI Tests

File: app/config/admin/security.yml
Comment (Hlavtox) [inline app/config/admin/security.yml:49]: @jolelievre Please make sure that this works with search cron, wouldn't it be blocked?

**Review by Hlavtox — CHANGES_REQUESTED** [review]
Does not seem to reimplement PS_COOKIE_CHECKIP removed. This feature allows to enable/disable checking IP address of the session.

Correct me and remove the block if I am wrong. :-)

File: src/PrestaShopBundle/Security/Admin/AdminAuthenticationSuccessHandler.php
Comment (nicosomb) [inline src/PrestaShopBundle/Security/Admin/AdminAuthenticationSuccessHandler.php:77]: Why don't initialize `$homeUrl` with `$this->legacyContext->getAdminLink('AdminDashboard');` ?

File: app/config/admin/security.yml
Comment (jolelievre) [reply app/config/admin/security.yml:49]: I didn't test it yet but it will likely fail, this has already been anticipated in the follow up tasks of this epic https://github.com/PrestaShop/PrestaShop/issues/36001

**Comment by jolelievre** [comment]
> Does not seem to reimplement PS_COOKIE_CHECKIP removed. This feature allows to enable/disable checking IP address of the session.
> 
> Correct me and remove the block if I am wrong. :-)

I didn't know about this one, just like the cron URLs feature I added this in the epic https://github.com/PrestaShop/PrestaShop/issues/36001

The present PR is already big enough but it still passes all our autotests, a few left tasks are already identified and will be handled in smaller PRs, if you have other thoughts about missing things feel free to let us know so that we can add them into the epic

I don't think we should block the PR though considering it already handles maybe 90% of the whole login/authorization feature and I can guarantee that all the remaining tasks will be dealt with right after this PR is merged 😉

File: admin-dev/index.php
Comment (M0rgan01) [inline admin-dev/index.php:81]: Finally :+1:

File: src/PrestaShopBundle/Security/Admin/AdminAuthenticationSuccessHandler.php
Comment (jolelievre) [reply src/PrestaShopBundle/Security/Admin/AdminAuthenticationSuccessHandler.php:77]: The home url is not necessarily the `AdminDashboard`, you can configure your own default page in the Employee profile, that's why it's dynamic and based on the employee's configuration

As such a new dedicated Symfony controller was implemented `admin_homepage` which is bound to the `/` URL and handles this dynamic redirection, so actually here we could use this `admin_homepage` route which is better suited than the redirection to `AdminDashboard`

But if e redirect to a route that is also only handling a redirection it's an extra request that we can avoid by redirecting to the appropriate URL here right after the login

One improvement though could be to extract this homepage URL fetching process in a dedicated server because this class and the `homepageAction` controller do have common and duplicated code I'll add this small refacto in the TO LIST in the epic https://github.com/PrestaShop/PrestaShop/issues/36001

File: app/config/admin/security.yml
Comment (M0rgan01) [reply app/config/admin/security.yml:49]: It is possible to redo the Search controller under Symfony, and give it public access via attribute isGranted. For later

File: app/config/admin/services.yml
Comment (M0rgan01) [inline app/config/admin/services.yml:30]: :clap:

File: app/config/admin/services.yml
Comment (M0rgan01) [inline app/config/admin/services.yml:26]: Not necessary due to default value

**Review by nicosomb — APPROVED** [review]

File: app/config/parameters.yml.dist
Comment (M0rgan01) [inline app/config/parameters.yml.dist:27]: We could put this in environment variables

**Review by Hlavtox — APPROVED** [review]

**Comment by Hlavtox** [comment]
@jolelievre No worries, I trust you. Let's merge this so the PR doesn't keep growing. I added the notes to https://github.com/PrestaShop/PrestaShop/issues/36001. ;-)

File: src/PrestaShopBundle/Entity/Employee/Employee.php
Comment (M0rgan01) [inline src/PrestaShopBundle/Entity/Employee/Employee.php:98]: Is it interesting to add a unique constraint to this field?

File: src/PrestaShopBundle/Entity/Employee/Employee.php
Comment (M0rgan01) [inline src/PrestaShopBundle/Entity/Employee/Employee.php:193]: When I see that I tell myself that it could go to another entity

Comment (Hlavtox) [reply src/PrestaShopBundle/Entity/Employee/Employee.php:193]: This is no longer used, maybe we can use this opportunity to remove this 💩?

bo_color
bo_width
bo_theme (there is only default)
bo_menu

Ping @kpodemski correct me if I miss something

Comment (jolelievre) [reply src/PrestaShopBundle/Entity/Employee/Employee.php:193]: I don't mind, for now I had to leave them in the Doctrine entity or the doctrine command would automatically delete these fields, so I had to sync the content of the entity with the db structure table

We could clean those, but in another PR then

File: src/PrestaShopBundle/Entity/Employee/Employee.php
Comment (jolelievre) [reply src/PrestaShopBundle/Entity/Employee/Employee.php:98]: Not for now, the Doctrine entity is not used in edition (not in the form nor in the CQRS commands), so they would be useless The should be added when we refacto the CQRS commands I guess

File: app/config/parameters.yml.dist
Comment (jolelievre) [reply app/config/parameters.yml.dist:27]: Yep for now they are default parameters, if we use env variables we can always use the `env` directive in the parameters, but currently we transform parameters in a PHP file so we can't really enjoy this feature To use this we should do a small refacto to actually use the `parameters.yml` file

I believe I had to add those default values here because they would make the SF console break when absent at some point, and it's more consistent to have all the required fields in the dist file anyway

File: app/config/admin/services.yml
Comment (jolelievre) [reply app/config/admin/services.yml:26]: I note to remove it in the next PR 😉

File: src/PrestaShopBundle/Entity/Employee/Employee.php
Comment (jolelievre) [reply src/PrestaShopBundle/Entity/Employee/Employee.php:98]: Argh my bad I misunderstood I thought you were talking about assertions for the entity validation Regarding the unicity of this column you're completely right But it's a change of the DB structure I prefer to handle it in another PR, we also discussed in another comment about removing the theme customizations field, I guess all this could be handled in a dedicated PR

Comment (M0rgan01) [reply src/PrestaShopBundle/Entity/Employee/Employee.php:98]: Ok :+1:

**Review by M0rgan01 — APPROVED** [review]

File: app/config/admin/security.yml
Comment (jolelievre) [reply app/config/admin/security.yml:49]: Yep but that's the easy way I guess 😅 we still have legacy controllers in modules that will need this feature

**Comment by paulnoelcholot** [comment]
Hello @jolelievre,

I tested your PR and it's good for me! :tada: 

Thanks!

**Review by paulnoelcholot — APPROVED** [review]

**Comment by gericfo** [comment]
Hello @jolelievre 
Please note that the **“Needs autoupgrade PR”** label has been automatically added to your Pull Request. This means that the changes made in your PR must also be reported in the **Autoupgrade module**, responsible for running the update process of PrestaShop stores. 
When there's a modification to the `db-structure.sql` file or if a Doctrine entity has been modified, added, or deleted, it impacts the database structure. Modifying the list of hooks and feature flags has consequences on the update process as well. These changes can affect how the application interacts with the database and how update procedures are handled. It's important to consider these aspects when contributing to ensure a smooth update experience for users. 
That's why this contribution should be backported, with another corresponding PR, on the **Autoupgrade module's GitHub repository** which is available here: [link](https://github.com/PrestaShop/autoupgrade). 
Don't forget to add a comment in this current thread with the link to this PR once it has been created. 
Thank you.

**Comment by gericfo** [comment]
**Related Autoupgrade PR** - [https://github.com/PrestaShop/autoupgrade/pull/954](https://github.com/PrestaShop/autoupgrade/pull/954)

**Pull request merged**
